### PR TITLE
added helm files for Directv

### PR DESCRIPTION
**Pull request by priya64646**
added helm files for Directv

### CODE REVIEW

**Review by olamiwhat — COMMENTED** [review]
I like the idea of creating helm charts for this but this repo and its workflows are probably not set up yet for apps outside of localfirst. I think to make this work where we start using this repo for apps outside of localfirst, you'll have to make modification to the githubworkflows and/or directory structure to accomodate, no? Unless we don't care about the workflows for now.

An easier way to get this done which was what I thought we were doing is to just add a `values.yml` to the directtv repo and reference that file in the harness deployment you'd create.

**Comment by priya64646** [comment]
```
pbommineni@Macadmins-MacBook-Pro charts % helm install directv  --dry-run --debug -f ~/values.yml  Directv
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/pbommineni/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/pbommineni/.kube/config
install.go:178: [debug] Original chart version: ""
install.go:195: [debug] CHART PATH: /Users/pbommineni/github/helm-charts/charts/Directv

NAME: directv
LAST DEPLOYED: Mon Mar  7 19:15:49 2022
NAMESPACE: cheddar
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
deployment:
  configmap__name: directv
  env:
    config:
      NODE_ENV: production
      REACT_APP_GRAPHQL_ENV: production
    secrets:
      names:
      - dtv-secrets
  main:
    image: ${artifact.metadata.image}
    name: cheddar-dtv
    promExporter:
      args:
      - -nginx.scrape-uri=http://localhost/stub_status
      image: nginx/nginx-prometheus-exporter:0.4.2
      name: nginx-prom-exporter
      ports:
        containerPort: 9113
        name: nginx-ex-port
      resources:
        limits:
          cpu: 100m
          memory: 100Mi
        requests:
          cpu: 100m
          memory: 50Mi
    replicas: 2
    resources:
      limits:
        cpu: 100m
        memory: 100Mi
      requests:
        cpu: 100m
        memory: 50Mi
  namespace: ${infra.kubernetes.namespace}

COMPUTED VALUES:
deployment:
  configmap__name: directv
  env:
    config:
      NODE_ENV: production
      REACT_APP_GRAPHQL_ENV: production
    secrets:
      names:
      - dtv-secrets
  main:
    image: ${artifact.metadata.image}
    name: cheddar-dtv
    promExporter:
      args:
      - -nginx.scrape-uri=http://localhost/stub_status
      image: nginx/nginx-prometheus-exporter:0.4.2
      name: nginx-prom-exporter
      ports:
        containerPort: 9113
        name: nginx-ex-port
      resources:
        limits:
          cpu: 100m
          memory: 100Mi
        requests:
          cpu: 100m
          memory: 50Mi
    replicas: 2
    resources:
      limits:
        cpu: 100m
        memory: 100Mi
      requests:
        cpu: 100m
        memory: 50Mi
  namespace: ${infra.kubernetes.namespace}

HOOKS:
MANIFEST:
---
# Source: Directv/templates/configMap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: directv
data:
  NODE_ENV: production
  REACT_APP_GRAPHQL_ENV: production
---
# Source: Directv/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: cheddar-dtv
  namespace: ${infra.kubernetes.namespace}
  labels:
    app: cheddar-dtv
spec:
  replicas: 2
  selector:
    matchLabels:
      app: cheddar-dtv
  template:
    metadata:
      labels:
        app: cheddar-dtv
    spec:
      containers:
      - name: cheddar-dtv
        image: ${artifact.metadata.image}
        envFrom:
        - configMapRef:
            name: cheddar-dtv
        - secretRef:
            name: dtv-secrets
        resources:
          limits:
            cpu: 100m
            memory: 100Mi
          requests:
            cpu: 100m
            memory: 50Mi
      - name: nginx-prom-exporter
        image: nginx/nginx-prometheus-exporter:0.4.2
        args:
          - "-nginx.scrape-uri=http://localhost/stub_status"
        ports:
        - name: nginx-ex-port
          containerPort: 9113
        resources:
          limits:
            cpu: 100m
            memory: 100Mi
          requests:
            cpu: 100m
            memory: 50Mi
```

**Comment by priya64646** [comment]
@olamiwhat , @pankajnarang I have added the dry run out put for directv helm charts. Please verify and approve

**Comment by pankajnarang** [comment]
Ports section in `cheddar-dtv` container is missing

**Comment by priya64646** [comment]
```
pbommineni@Macadmins-MacBook-Pro charts % helm install directv  --dry-run --debug -f ~/values.yml  Directv
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/pbommineni/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/pbommineni/.kube/config
install.go:178: [debug] Original chart version: ""
install.go:195: [debug] CHART PATH: /Users/pbommineni/github/helm-charts/charts/Directv

NAME: directv
LAST DEPLOYED: Tue Mar  8 11:31:01 2022
NAMESPACE: cheddar
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
deployment:
  configmap__name: directv
  env:
    config:
      NODE_ENV: production
      REACT_APP_GRAPHQL_ENV: production
    secrets:
      names:
      - dtv-secrets
  main:
    image: ${artifact.metadata.image}
    name: cheddar-dtv
    ports:
      containerPort: 80
      name: http
    promExporter:
      args:
      - -nginx.scrape-uri=http://localhost/stub_status
      image: nginx/nginx-prometheus-exporter:0.4.2
      name: nginx-prom-exporter
      ports:
        containerPort: 9113
        name: nginx-ex-port
      resources:
        limits:
          cpu: 100m
          memory: 100Mi
        requests:
          cpu: 100m
          memory: 50Mi
    replicas: 2
    resources:
      limits:
        cpu: 100m
        memory: 100Mi
      requests:
        cpu: 100m
        memory: 50Mi
  namespace: ${infra.kubernetes.namespace}

COMPUTED VALUES:
deployment:
  configmap__name: directv
  env:
    config:
      NODE_ENV: production
      REACT_APP_GRAPHQL_ENV: production
    secrets:
      names:
      - dtv-secrets
  main:
    image: ${artifact.metadata.image}
    name: cheddar-dtv
    ports:
      containerPort: 80
      name: http
    promExporter:
      args:
      - -nginx.scrape-uri=http://localhost/stub_status
      image: nginx/nginx-prometheus-exporter:0.4.2
      name: nginx-prom-exporter
      ports:
        containerPort: 9113
        name: nginx-ex-port
      resources:
        limits:
          cpu: 100m
          memory: 100Mi
        requests:
          cpu: 100m
          memory: 50Mi
    replicas: 2
    resources:
      limits:
        cpu: 100m
        memory: 100Mi
      requests:
        cpu: 100m
        memory: 50Mi
  namespace: ${infra.kubernetes.namespace}

HOOKS:
MANIFEST:
---
# Source: Directv/templates/configMap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: directv
data:
  NODE_ENV: production
  REACT_APP_GRAPHQL_ENV: production
---
# Source: Directv/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: cheddar-dtv
  namespace: ${infra.kubernetes.namespace}
  labels:
    app: cheddar-dtv
spec:
  replicas: 2
  selector:
    matchLabels:
      app: cheddar-dtv
  template:
    metadata:
      labels:
        app: cheddar-dtv
    spec:
      containers:
      - name: cheddar-dtv
        image: ${artifact.metadata.image}
        envFrom:
        - configMapRef:
            name: cheddar-dtv
        - secretRef:
            name: dtv-secrets
        resources:
          limits:
            cpu: 100m
            memory: 100Mi
          requests:
            cpu: 100m
            memory: 50Mi
        ports:
        - name: http
          containerPort: 80
      - name: nginx-prom-exporter
        image: nginx/nginx-prometheus-exporter:0.4.2
        args:
          - "-nginx.scrape-uri=http://localhost/stub_status"
        ports:
        - name: nginx-ex-port
          containerPort: 9113
        resources:
          limits:
            cpu: 100m
            memory: 100Mi
          requests:
            cpu: 100m
            memory: 50Mi
```

**Review by olamiwhat — CHANGES_REQUESTED** [review]
LGTM. Just the cronjob template, once we remove that it should be good to merge.

File: charts/Directv/templates/cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+{{- if .Values.deployment.env.cronjob }}
```
[the remark this answers is not in the document]
Comment (pankajnarang) [reply charts/Directv/templates/cronjob.yaml:1]: Its not required but is standard with all chart releases that we do. So next time when we need this, we just have to update the values file and won't have to release the chart again.

**Review by olamiwhat — APPROVED** [review]

**Review by pankajnarang — APPROVED** [review]

**Pull request merged**